### ISSUE TITLE
How to get rdm after EOM?

### ISSUE THREAD

**Issue by sunnowrun** [MilaimKas/ECW_CC#1]
hello, I am looking for method to get reduced density matrix after EOM. And I find in your code ECW_CC)/gamma_exp.py there is a class ESexp say can do such things. But I didn't find the corresponding code, is it still not completed? Do you know how to realize it?
thanks in advance!

**Comment by MilaimKas** [comment]
Hi there,
I have implemented the EOM rdm1 for the single CC case (CCS-EOM(1)). For general purposes, you should use the CCSD-EOM(2) rdm1. This has not yet been implemented in PySCF.
Take a look at the following issue: https://github.com/pyscf/pyscf/issues/632. Someone posted a solution for the EOM-IP case.

**Comment by sunnowrun** [comment]
Thank you.
Yes, I found that issue before. But I need a solution for EOM-EE case, while that code need left-eigenvector, it seems that left one is not realized in PySCF. So, I don't know how to figure it out.
Do you know how to realize it? or how about theoretically reference?

**Comment by MilaimKas** [comment]
Yes exactly, right and left eigenvectors (r and l amplitudes) for the excited states, in addition to the t and lambda amplitudes for the ground state, are needed for property calculation through the reduced density matrix.
And again you are right, the equation for the left eigenvector are not yet implemented in PySCF.

I think the best way to start is to look at [Bartlett's book  ](https://doi.org/10.1063/1.464746) and the reference therein.

Good luck

**Comment by sunnowrun** [comment]
OK, I will read it. Thank you!
